sdk/communication/communication-short-codes/src/utils/communicationAccessKeyCredentialPolicy.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isTokenCredential, KeyCredential, TokenCredential } from "@azure/core-auth";
+import {
+  PipelinePolicy,
+  PipelineRequest,
+  bearerTokenAuthenticationPolicy,
+} from "@azure/core-rest-pipeline";
+import { URL } from "./url";
+import { shaHash, shaHMAC } from "./cryptoUtils";
+export const communicationAccessKeyCredentialPolicyName = "CommunicationAccessKeyCredentialPolicy";
+
+/**
+ * Creates an HTTP pipeline policy to authenticate a request using a `KeyCredential`.
+ *
+ * @param credential - The key credential.
+ */
+export const createCommunicationAccessKeyCredentialPolicy = (
```

Comment:
we need to wait for this to get merged and released: https://github.com/Azure/azure-sdk-for-js/pull/20337